### PR TITLE
Syntax highlighting for Kotlin DSL files

### PR DESCRIPTION
**Pull request by D0zee**

### CODE REVIEW

File: org.eclipse.buildship.kotlinDSL.feature/category.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<site>
+   <bundle id="org.eclipse.buildship.kotlinDSL.provider">
```
Comment (donat) [inline org.eclipse.buildship.kotlinDSL.feature/category.xml:3]: same here

File: org.eclipse.buildship.kotlinDSL.provider/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Provider
+Bundle-SymbolicName: org.eclipse.buildship.kotlinDSL.provider;singleton:=true
```
Comment (donat) [inline org.eclipse.buildship.kotlinDSL.provider/META-INF/MANIFEST.MF:4]: same

File: org.eclipse.buildship.kotlinDSL.provider/build.gradle [hunk 3]
```diff
@@ -0,0 +1,7 @@
+plugins {
+    id 'application'
+}
+apply plugin: eclipsebuild.BundlePlugin
+repositories {
```
Comment (donat) [inline org.eclipse.buildship.kotlinDSL.provider/build.gradle:5]: ❓ Why do you need to declare a repository if you don't declare any dependencies?

File: org.eclipse.buildship.kotlinDSL.provider/build.gradle [hunk 4]
```diff
@@ -0,0 +1,7 @@
+plugins {
+    id 'application'
+}
```
Comment (donat) [inline org.eclipse.buildship.kotlinDSL.provider/build.gradle:3]: This is an eclipse plugin project; you don't need to apply the application plugin,

```suggestion
```

File: org.eclipse.buildship.kotlinDSL.provider/src/syntax/Kotlin.tmLanguage.json [hunk 5]
```diff
@@ -0,0 +1,569 @@
+{
```
Comment (donat) [inline org.eclipse.buildship.kotlinDSL.provider/src/syntax/Kotlin.tmLanguage.json:1]: ❌ imho this file should be in the `src/main/resources` directory.

**Review by bitcoinfinancier — APPROVED** [review]

**Review by donat — APPROVED** [review]

**Pull request merged**